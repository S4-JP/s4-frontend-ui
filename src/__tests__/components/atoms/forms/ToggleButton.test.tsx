import React from 'react'
import { render } from '@testing-library/react'
import { ToggleButton, Color, Icon } from '../../../../index'

/**
 * type Props = {
 *   checked?: boolean
 *   onChange?: (e: React.SyntheticEvent) => void
 *   id?: string
 *   value?: string
 *   name?: string
 *   isInvalid?: boolean
 *   handleClick?(e: React.MouseEvent<HTMLElement>): void
 * }
 */
describe('form - ToggleButton のレンダリング確認', () => {
  test('オプションなしで設定した場合、初期値でレンダリングされる。', () => {
    const { getByTestId } = render(<ToggleButton />)

    expect(getByTestId('atoms-f-toggle-button-label')).toMatchSnapshot()
    expect(getByTestId('atoms-f-toggle-button-marker')).toHaveStyle(
      `background: ${Color.COMPONENT.TOGGLE};`
    )
    expect(getByTestId('atoms-f-toggle-button-on')).toHaveStyle(`opacity: 0;`)
    expect(getByTestId('atoms-f-toggle-button-off')).toHaveStyle(`opacity: 1;`)
  })

  test('checkedにtrueを設定した場合、Makerの背景色が設定される。', () => {
    const { getByTestId } = render(
      <ToggleButton
        checked={true}
        onChange={jest.fn()}
        handleClick={jest.fn()}
      />
    )

    expect(getByTestId('atoms-f-toggle-button-marker')).toHaveStyle(
      `background: ${Color.PRIMARY._500};`
    )
    expect(getByTestId('atoms-f-toggle-button-on')).toHaveStyle(`opacity: 1;`)
    expect(getByTestId('atoms-f-toggle-button-off')).toHaveStyle(`opacity: 0;`)
  })
})
