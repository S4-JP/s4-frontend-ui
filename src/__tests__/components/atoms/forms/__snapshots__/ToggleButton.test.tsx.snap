// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form - ToggleButton のレンダリング確認 オプションなしで設定した場合、初期値でレンダリングされる。 1`] = `
<label
  class="sc-iIPllB gQpSmt"
  data-testid="atoms-f-toggle-button-label"
>
  <div
    class="sc-gicCDI ihZzBr"
    data-testid="atoms-f-toggle-button-marker"
  >
    <p
      class="sc-ezWOiH dHdGUL"
      data-testid="atoms-f-toggle-button-on"
    >
      ON
    </p>
    <p
      class="sc-bZkfAO yACOH"
      data-testid="atoms-f-toggle-button-off"
    >
      OFF
    </p>
  </div>
  <input
    class="sc-kLLXSd bNKuxI"
    data-testid="atoms-f-toggle-button-inner-input"
    type="checkbox"
  />
</label>
`;
